test/k8sT/KafkaPolicies.go

```diff
@@ -0,0 +1,257 @@
+		for _, v := range apps {
+			fmt.Printf("getting pod name of app: %s\n", v)
+			res, err := kubectl.GetPodNames(helpers.DefaultNamespace, fmt.Sprintf("app=%s", v))
+			Expect(err).Should(BeNil())
+			Expect(res).Should(Not(BeNil()))
+			fmt.Printf("res output: %q\n", res)
+			appPods[v] = res[0]
+			logger.Infof("KafkaPolicyRulesTest: pod=%q assigned to %q", res[0], v)
+		}
+		return appPods
+	}
+
+	It("KafkaPolicies", func() {
+		clusterIP, err := kubectl.Get(helpers.DefaultNamespace, "svc").Filter(
+			"{.items[?(@.metadata.name == \"kafka-service\")].spec.clusterIP}")
+		logger.Infof("PolicyRulesTest: cluster service ip '%s'", clusterIP)
+		Expect(err).Should(BeNil())
+
+		ciliumPod, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+		Expect(err).Should(BeNil())
+
+		status := kubectl.CiliumExec(ciliumPod, fmt.Sprintf("cilium config %s=%s", helpers.PolicyEnforcement, helpers.PolicyEnforcementDefault))
+		status.ExpectSuccess()
+		helpers.Sleep(5)
+
+		kubectl.CiliumEndpointWait(ciliumPod)
+
+		epsStatus := helpers.WithTimeout(func() bool {
+			endpoints, err := kubectl.CiliumEndpointsListByLabel(ciliumPod, podFilter)
+			if err != nil {
+				return false
+			}
+			return endpoints.AreReady()
+		}, "Could not get endpoints", &helpers.TimeoutConfig{Timeout: 100})
+		Expect(epsStatus).Should(BeNil())
+
+		endpoints, err := kubectl.CiliumEndpointsListByLabel(ciliumPod, podFilter)
+		Expect(err).Should(BeNil())
+
+		Expect(endpoints.AreReady()).Should(BeTrue())
+
+		appPods := getAppPods()
+		By("Testing basic kafka setup")
+
+		_, err = kubectl.ExecKafkaPodCmd(
+			helpers.DefaultNamespace, appPods[helpers.EmpireHq], fmt.Sprintf("-c \"echo “Happy 40th Birthday to General Tagge” | ./kafka-produce.sh --topic empire-announce\""))
+		Expect(err).Should(BeNil())
+
+		_, err = kubectl.ExecKafkaPodCmd(
+			helpers.DefaultNamespace, appPods[helpers.Outpost], fmt.Sprintf("-c \"./kafka-consume.sh --topic empire-announce --from-beginning --max-messages 1\""))
+		Expect(err).Should(BeNil())
+
+		_, err = kubectl.ExecKafkaPodCmd(
+			helpers.DefaultNamespace, appPods[helpers.EmpireHq], fmt.Sprintf("-c \"echo “deathstar reactor design v3” | ./kafka-produce.sh --topic deathstar-plans\""))
+		Expect(err).Should(BeNil())
+
+		_, err = kubectl.ExecKafkaPodCmd(
+			helpers.DefaultNamespace, appPods[helpers.Outpost], fmt.Sprintf("-c \"./kafka-consume.sh --topic deathstar-plans --from-beginning --max-messages 1\""))
+		Expect(err).Should(BeNil())
+
+		_, err = kubectl.ExecKafkaPodCmd(
+			helpers.DefaultNamespace, appPods[helpers.Backup], fmt.Sprintf("-c \"echo “Happy 40th Birthday to General Tagge” | ./kafka-produce.sh --topic empire-announce\""))
+		Expect(err).Should(BeNil())
+
+		_, err = kubectl.ExecKafkaPodCmd(
+			helpers.DefaultNamespace, appPods[helpers.Outpost], fmt.Sprintf("-c \"echo “Vader Booed at Empire Karaoke Party” | ./kafka-produce.sh --topic empire-announce\""))
```

Comment:
Can you create wrappers around these `kafka-produce` and `kafka-consume` commands? There is lots of hardcoding of strings which can be factored into little helper functions.